### PR TITLE
Add systempath grain

### LINKED ISSUE #49049: Grains: path (string) vs. pythonpath (array)
### Description of Issue/Question
Why is the *path* grain a string while *pythonpath* is an array? Could this be streamlined, please (preferred: array, maybe bound to a config option to stay backwards compatible)? This is quite annoying since one always needs to take OS differences (':' vs. ';') into account.

### Setup
```
# salt myminion grains.item path pythonpath
myminion:
    ----------
    path:
        /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    pythonpath:
        - /usr/bin
        - /usr/lib/python2.7
        - /usr/lib/python2.7/plat-x86_64-linux-gnu
        - /usr/lib/python2.7/lib-tk
        - /usr/lib/python2.7/lib-old
        - /usr/lib/python2.7/lib-dynload
        - /usr/local/lib/python2.7/dist-packages
        - /usr/lib/python2.7/dist-packages
```
### Steps to Reproduce Issue
Get both grains as shown above.

### Versions Report
```
# salt --versions-report
Salt Version:
           Salt: 2018.3.2

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Dec  4 2017, 14:50:18)
   python-gnupg: 0.3.8
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-1062-aws
         system: Linux
        version: Ubuntu 16.04 xenial
```

### PR DESCRIPTION
**Pull request by terminalmage**
This provides a grain like pythonpath, only for the system path. In
other words, it's the value of the PATH environment variable, only
formatted as a list of directories.

Resolves #49049.

### CODE REVIEW

**Review by DmitryKuzmenko — DISMISSED** [review]

**Comment by terminalmage** [comment]
I rebased to resolve merge conflicts.

**Comment by terminalmage** [comment]
@DmitryKuzmenko When I rebased, it looks like I lost your review. I can't find the old review comments, either. Did you have any concerns.

One thing to note is that I used the wrong name in the Pull Request title, it should have been `systempath` instead of `systemtest`. I have renamed the PR to reflect this.

**Review by DmitryKuzmenko — DISMISSED** [review]

**Comment by DmitryKuzmenko** [comment]
@terminalmage if I remember correctly I've just approved this. =)

**Comment by terminalmage** [comment]
OK, the GitHub web interface was showing that your earlier review was a "comment" review instead of an approval, so I wanted to be sure.

... and there is now another merge conflict, so I will have to rebase again.

**Comment by terminalmage** [comment]
Oh great... guess most of my PRs are going to need a rebase now that this harebrained quest to use black has been pushed through.

Good to see that the important things are getting done... 🙄

**Comment by DmitryKuzmenko** [comment]
Yep, unfortunately.

**Comment by terminalmage** [comment]
Looks like you'll need to re-review, again.

File: tests/unit/grains/test_core.py [hunk 1]
```diff
@@ -1784,3 +1784,10 @@ def test_saltversioninfo(self):
             assert len(info) == 2
         assert all([x is not None for x in info])
         assert all([isinstance(x, int) for x in info])
+
+    def test_path(self):
+        comps = ['foo', 'bar', 'baz']
```
Comment (DmitryKuzmenko) [inline tests/unit/grains/test_core.py:1789]: Double-quotes here please.

**Review by DmitryKuzmenko — DISMISSED** [review]

**Review by dwoz — APPROVED** [review]

**Pull request merged**